workalendar_holidays/wizards/workalendar_holiday_import.py

```diff
@@ -0,0 +1,146 @@
+    interval_type = fields.Selection([
+        ('days', 'Day(s)'),
+        ('weeks', 'Week(s)'),
+        ('months', 'Month(s)'),
+        ('years', 'Year(s)')],
+        string='Type', default='days', required=True)
+    calendar_id = fields.Many2one(
+        'resource.calendar', string="Work Time")
+    country_group_id = fields.Many2one(
+        'res.country.group', string="Continent", required=True)
+    country_id = fields.Many2one(
+        'res.country', string="Country", required=True)
+
+    @api.model
+    def _get_workalendar_module(self, continent_name, country_id=False):
+        country_libs = {
+            self.env.ref('base.ca').id: canada,
+            self.env.ref('base.us').id: usa
+        }
+        if country_id in country_libs:
+            return country_libs[country_id]
+        name = str(continent_name).lower()
+        libs = [africa, america, asia, europe, oceania]
+        module = filter(lambda x: (x.__name__).split('.')[1] == name, libs)
+        return module and module[0] or False
+
+    @api.model
+    def _get_workalendar_instance(self, continent, country, state=False):
+        instance = False
+        module = self._get_workalendar_module(continent.name)
+        if not module:
+            raise ValidationError(
+                _("Sorry, no support for %s continent") % continent.name)
+
+        # TODO: at version 9.0 can be replace the following search with the
+        # ir.translation function _get_terms_query
+        trans = self.env['ir.translation'].search([
+            ('type', '=', 'model'),
+            ('name', '=', 'res.country,name'),
+            ('res_id', '=', country.id)], limit=1)
+        country_name = trans.src and \
+            str(''.join(trans.src.title().split(' '))) or country.name
+        try:
+            class_ = getattr(module, country_name)
+            instance = class_()
+        except AttributeError:
+            raise ValidationError(
+                _("Sorry, no support for country: %s") % country.name)
+        return instance
+
+    @api.multi
+    def holiday_import(self):
+        for wiz in self:
+            leaves = self.env['resource.calendar.leaves']
+            work_time = wiz.calendar_id
+            continent = wiz.country_group_id
+            country = wiz.country_id
+            calendar = self._get_workalendar_instance(continent, country)
+            start_year = fields.Datetime.from_string(wiz.start_date).year
+            end_year = fields.Datetime.from_string(wiz.end_date).year
+            public_holidays = []
+            for year in range(start_year, end_year + 1):
+                public_holidays += calendar.holidays(year)
+            for public_holiday in public_holidays:
+                holiday_date, holiday_name = public_holiday
+
+                # jump when the public holiday date is above wizard end date
+                if holiday_date > fields.Date.from_string(wiz.end_date) or \
```

Comment:
you can validate this in the line [117](https://github.com/Vauxoo/addons-vauxoo/pull/1077/files#diff-f499a7274fa16e2a9b2b6c80f2d6dd85R117),  to avoid add an unneeded value in the list.